src/otg/eqtl_catalogue.py

```diff
@@ -0,0 +1,67 @@
+"""Step to run eQTL Catalogue study table ingestion."""
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+
+from omegaconf import MISSING
+
+from otg.common.session import Session
+from otg.datasource.eqtl_catalogue.study_index import EqtlCatalogueStudyIndex
+from otg.datasource.eqtl_catalogue.summary_stats import EqtlCatalogueSummaryStats
+
+
+@dataclass
+class EqtlCatalogueStep:
+    """eQTL Catalogue ingestion step.
+
+    Attributes:
+        session (Session): Session object.
+        eqtl_catalogue_paths_imported (str): eQTL Catalogue input files for the harmonised and imported data.
+        eqtl_catalogue_study_index_out (str): Output path for the eQTL Catalogue study index dataset.
+        eqtl_catalogue_summary_stats_out (str): Output path for the eQTL Catalogue summary stats.
+    """
+
+    session: Session = MISSING
+
+    eqtl_catalogue_paths_imported: str = MISSING
+    eqtl_catalogue_study_index_out: str = MISSING
+    eqtl_catalogue_summary_stats_out: str = MISSING
+
+    def __post_init__(self: EqtlCatalogueStep) -> None:
+        """Run step."""
+        # Fetch study index.
+        df = self.session.spark.read.option("delimiter", "\t").csv(
+            self.eqtl_catalogue_paths_imported, header=True
+        )
+        # Process partial study index.  At this point, it is not complete because we don't have the gene IDs, which we
+        # will only get once the summary stats are ingested.
+        study_index_df = EqtlCatalogueStudyIndex.from_source(df).df
+
+        # Fetch summary stats.
+        input_filenames = [row.summarystatsLocation for row in study_index_df.collect()]
```

Comment:
At this specific stage of the pipeline, the `study_index_df.collect()` call is actually tiny, because it's before the study index expansion — it only contains the original 49 rows from the TSV file provided by the eQTL Catalogue.

I did benchmark adding the schema, but it only shaves off about 15–30 seconds for both FinnGen and the eQTL Catalogue. It looks like, with our current implementation where Spark has to ingest very large partitions (again, both for FinnGen and eQTL Catalogue), schema detection isn't a limiting step

Reply:
Oh, yes, this is the studies.